os/arch/arm/src/imxrt/imxrt_pit.c

```diff
@@ -151,3 +216,381 @@ uint64_t imxrt_pit_getlifetimetimercount(PIT_Type *base)
+
+	imxrt_pit_clearstatusflags(pit->base, priv->channel, kPIT_TimerFlag);
+
+	if (priv->callback(&next_interval_us, priv->arg)) {
+		if (next_interval_us > 0) {
+			freq = imxrt_clock_getperclkfreq();
+			ticks = ((uint64_t)freq * (uint64_t)next_interval_us) / USEC_PER_SEC;
+			imxrt_pit_settimerperiod(priv->pit->base, priv->channel, ticks);
+		}
+	} else {
+		imxrt_pit_stop((struct timer_lowerhalf_s *)priv);
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: imxrt_pit_start
+ *
+ * Description:
+ *   Start the timer, resetting the time to the current timeout,
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int imxrt_pit_start(struct timer_lowerhalf_s *lower)
+{
+	struct imxrt_pit_lowerhalf_s *priv = (struct imxrt_pit_lowerhalf_s *)lower;
+
+	if (!priv->started) {
+		if (priv->callback != NULL) {
+			imxrt_pit_enableinterrupts(priv->pit->base, priv->channel, kPIT_TimerInterruptEnable);
+		}
+
+		if (active == 0) {
+			imxrt_pit_setisr(priv->pit, imxrt_pit_handler, priv);
+		}
+
+		if (ch_status[priv->channel])
+			imxrt_pit_starttimer(priv->pit->base, (priv->channel-1));
+
+		imxrt_pit_starttimer(priv->pit->base, priv->channel);
+
+		priv->started = true;
+		busy_status[priv->channel] = true;
+		if (ch_status[priv->channel]) {
+			busy_status[priv->channel-1] = true;
+		}
+
+		active++;
+
+		tmrvdbg("PIT Timer Channel %d is started, callback %s\n", priv->channel, priv->callback == NULL ? "none" : "set");
+		return OK;
+	}
+
+	/* Return EBUSY to indicate that the timer was already running */
+
+	tmrdbg("Error!! PIT Timer %d was already running\n", priv->channel);
+	return -EBUSY;
+}
+
+/****************************************************************************
+ * Name: imxrt_pit_stop
+ *
+ * Description:
+ *   Stop the timer
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int imxrt_pit_stop(struct timer_lowerhalf_s *lower)
+{
+	struct imxrt_pit_lowerhalf_s *priv = (struct imxrt_pit_lowerhalf_s *)lower;
+
+	if (priv->started) {
+		imxrt_pit_disableinterrupts(priv->pit->base, priv->channel, kPIT_TimerInterruptEnable);
+
+		imxrt_pit_stoptimer(priv->pit->base, priv->channel);
+		if (ch_status[priv->channel]) {
+			imxrt_pit_stoptimer(priv->pit->base, priv->channel-1);
+			imxrt_pit_settimerchainmode(priv->pit->base, priv->channel, false);
+			ch_status[priv->channel] = false;
+			busy_status[priv->channel-1] = false;
+		}
+
+		if (active == 1) {

```

Comment:
That's why I mentioned one more lowerhalf to control each of channels. you are counting another dev's status here. All lower-half should work independently.

Reply:
Done